sklearn/utils/stats.py

```diff
@@ -1,9 +1,10 @@
 # Authors: The scikit-learn developers
 # SPDX-License-Identifier: BSD-3-Clause
 
-import numpy as np
-
-from .extmath import stable_cumsum
+from ..utils._array_api import (
+    _find_matching_floating_dtype,
+    get_namespace_and_device,
+)
 
 
 def _weighted_percentile(array, sample_weight, percentile_rank=50):
```

Comment:
For the sake of consistency with private functions that accept array API inputs, we should add `xp=None` here (and inspect the device of `array` internally).

Similarly, `_averaged_weighted_percentile` should also accept an `xp` argument and forward it to the two nested calls.

Reply:
Done.